src/getShortcuts.ts

```diff
@@ -1,44 +1,66 @@
+    const usedShortcuts = getExtensionSetting('usedShortcuts')
+    if (usedShortcuts === 'disable') return 
+    const usedShortcutsMode = getExtensionSetting('usedShortcuts.mode')
     const currentLine = document.lineAt(position.line)
-    if (currentLine.text.includes(':')) return
-    return rules
-        .filter(rule => typeof rule[0] === 'string' && typeof rule[1] === 'object')
-        .map(([shortcut, rule]) => {
-            const cssRules = (Array.isArray(rule) ? rule : Object.entries(rule))
-                .filter(([prop, value], i, rulesArr) => {
-                    if (getExtensionSetting('skipVendorPrefix') === 'none') return true
-                    const hasUnvendoredRule = (current: string, matchUnvendored: (vendorLength: number, unvendored: string) => boolean) => {
-                        const match = /^-\w+-/.exec(current)
-                        if (!match) return false
-                        return matchUnvendored(match[0]!.length, current.slice(match[0]!.length))
-                    }
 
-                    return !(
-                        hasUnvendoredRule(prop, (vendorLength, unvendored) => rulesArr.some(([rule]) => unvendored === rule.slice(vendorLength)))
-                        // hasUnvendored(value as string, match => rulesArr.some(([, value]) => (value as string).slice(match[0]!.length)))
-                    )
-                })
-                .map(([prop, value]) => {
-                    if (typeof value === 'number') value = `${value.toString()}px`
-                    return `${prop}: ${value!};`
-                })
-                .join('\n')
-            const label = shortcut as string
-            return {
-                label,
-                insertText: cssRules,
-                documentation: new vscode.MarkdownString().appendCodeblock(
-                    `.${label} {\n${cssRules
-                        .split('\n')
-                        .map(rule => ' '.repeat(2) + rule)
-                        .join('\n')}\n}`,
-                    'css',
-                ),
-                kind: vscode.CompletionItemKind.Event,
-            }
-        })
+    // TODO measure parsing time on big stylesheets
+    const { usedRules } = parseCss(document.getText(), document.offsetAt(position))
+    if (!/^\s*(\w|-)*\s*$/.test(currentLine.text)) return
+    const usedRulesOffsets = new Map<string, number>()
+    return compact(
+        rules
+            .filter(rule => typeof rule[0] === 'string' && typeof rule[1] === 'object')
+            .map(([shortcut, rule]): vscode.CompletionItem | undefined => {
+                const cssRulesArr = (Array.isArray(rule) ? rule : Object.entries(rule))
+                    .filter(([prop, value], i, rulesArr) => {
+                        if (usedShortcutsMode === 'only-rule' ? usedRules.get(prop) : usedRules.get(prop)?.value === value)
+                            usedRulesOffsets.set(shortcut as string, usedRules.get(prop)!.offset)
```

Comment:
As `return false` was removed, now we don't filter out declarations that were already used. Take `sr-only` as an example:
```scss
.sr-only {
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    clip: rect(0,0,0,0);
    white-space: nowrap;
    border-width: 0px;
}
```
What if we already used `position: absolute;` above? I don't think it makes sense to duplicate the declaration. Instead, we can show in documentation as comment, that it was used on some line. Example:
```scss
.sr-only {
    position: absolute; // L50
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    clip: rect(0,0,0,0);
    white-space: nowrap;
    border-width: 0px;
}
```
We should strikethrough or remove the suggestion only if all declarations of the shortcuts was used. In case if we strikethrough, insertText should insert all declarations of the suggestion.

Reply:
WDYT? Should be easy with suggestion-scope Map like `rule index - used offset` Then we can write something like this:
```ts
const usedShortcut = usedRules.size === cssRulesArr.length
```
No need to return `return false`. Of course, it's not the point where we should put a lot of effort, since 99% shorcuts contain only one declaration.